src/spi/log/ProgressLog.ts

```diff
@@ -52,3 +52,32 @@ export interface ProgressLog {
 }
 
 export type LogFactory = (name: string) => Promise<ProgressLog>;
+
+export class StringCapturingProgressLog implements ProgressLog {
```

Comment:
Let's put this in its own file, as @claymccoy and I split out all the other implementations into their own files.

Reply:
Done